infra/src/main/kotlin/fi/fta/geoviite/api/tracklayout/v1/ExtLocationTrackServiceV1.kt

```diff
@@ -0,0 +1,222 @@
+                )
+
+        return ExtLocationTrackResponseV1(
+            trackNetworkVersion = publication.uuid,
+            coordinateSystem = coordinateSystem,
+            locationTrack =
+                getExtLocationTrack(
+                    oid,
+                    locationTrack,
+                    layoutContext,
+                    publication.publicationTime,
+                    coordinateSystem,
+                    lang,
+                ),
+        )
+    }
+
+    fun createLocationTrackModificationResponse(
+        oid: Oid<LocationTrack>,
+        modificationsFromVersion: Uuid<Publication>,
+        trackNetworkVersion: Uuid<Publication>?,
+        coordinateSystem: Srid,
+    ): ExtModifiedLocationTrackResponseV1? {
+        val layoutContext = MainLayoutContext.official
+        val lang = LocalizationLanguage.FI
+
+        val fromPublication =
+            publicationDao.fetchPublicationByUuid(modificationsFromVersion)
+                ?: throw ExtTrackNetworkVersionNotFound("modificationsFromVersion=${modificationsFromVersion}")
+
+        val toPublication =
+            trackNetworkVersion?.let { uuid ->
+                publicationDao.fetchPublicationByUuid(uuid)
+                    ?: throw ExtTrackNetworkVersionNotFound("trackNetworkVersion=${uuid}")
+            } ?: publicationDao.fetchLatestPublications(LayoutBranchType.MAIN, count = 1).single()
+
+        return if (fromPublication == toPublication) {
+            logger.info(
+                "there cannot be any differences if the requested publication ids are the same publication=${fromPublication.id}"
+            )
+            null
+        } else {
+            val locationTrackId =
+                locationTrackDao.lookupByExternalId(oid)?.id
+                    ?: throw ExtOidNotFoundExceptionV1("location track lookup failed, oid=$oid")
+
+            val fromLocationTrackVersion =
+                locationTrackDao.fetchOfficialVersionAtMoment(
+                    layoutContext.branch,
+                    locationTrackId,
+                    fromPublication.publicationTime,
+                )
+                    ?: throw ExtLocationTrackNotFoundExceptionV1(
+                        "'from' version fetch failed, moment=${fromPublication.publicationTime}, locationTrackId=$locationTrackId"
+                    )
+
+            val toLocationTrackVersion =
+                locationTrackDao.fetchOfficialVersionAtMoment(
+                    layoutContext.branch,
+                    locationTrackId,
+                    toPublication.publicationTime,
+                )
+                    ?: throw ExtLocationTrackNotFoundExceptionV1(
+                        "'to' version fetch failed, moment=${toPublication.publicationTime}, locationTrackId=$locationTrackId"
+                    )
+
+            return if (fromLocationTrackVersion == toLocationTrackVersion) {
```

Comment:
Mielestäni tätä `return`:ia ei pitäisi tarvita, kun tämä on jo riviltä 122 alkavan `return`:in sisällä